### PR TITLE
Upgrade to Rails 5.1 and Mongoid 6.

### PR DESCRIPTION
**Pull request by danielroseman**
Mainly configuration updates and fixes for deprecations.

### CODE REVIEW

**Comment by floehopper** [comment]
@danielroseman: I haven't looked at this carefully, but I'm pretty sure you'll need most of the changes in alphagov/manuals-publisher#1154 if you want to be able to run migrations. Otherwise you'll see an error like this:

    NoMethodError: undefined method `alias_method_chain' for #<Class:MyMigration>

**Comment by danielroseman** [comment]
Thanks @floehopper - I've updated to use the fork as suggested.

File: config/puma.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Puma can serve each request in a thread from an internal thread pool.
```
Comment (tijmenb) [inline config/puma.rb]: Should we skip the puma config here?

**Comment by danielroseman** [comment]
Thanks @tijmenb - config removed. I've also rebased onto master and force-pushed.

**Pull request merged**